File: Arrays/ArrOperations.java
package Arrays;

import java.util.Scanner;

class ArrayOperations {
    Scanner sc = new Scanner(System.in);

    public void read(int[] arr, int n) {
        System.out.print("Enter the array elements : ");
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();
    }

    // function to find the largest element in a given array
    public int LargestElement(int[] arr, int n) {
        int largest = arr[0];
        for (int i = 1; i < n; i++) {
            if (arr[i] > largest)
                largest = arr[i];
        }
        return largest;
    }

    // function to find the second largest from a given array
    public int secondLargest(int[] arr, int n) {
        int largest = LargestElement(arr, n);

        int secLargest = -1;
        for (int i = 0; i < n; i++) {
            if (arr[i] > secLargest && arr[i] != largest) {
                secLargest = arr[i];
            }
        }
        return secLargest;
    }
}

public class LargestEle {
    public static void main(String[] args) {
        ArrayOperations Arr = new ArrayOperations();
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size of an array : ");
        int n = sc.nextInt();
        int arr[] = new int[n];
        int choice;
        do {
            System.out.println("\nMENU\n");
            System.out.println("1.To read the elements ");
            System.out.println("2.Largest element in the array");
            System.out.println("3.Second Largest element in the array");
            System.out.println("4.Exit");

            System.out.println("Enter your choice : ");
            choice = sc.nextInt();

            switch (choice) {
                case 1:
                    Arr.read(arr, n);
                    break;
                case 2:
                    System.out.println("The largest element in the array: " + Arr.LargestElement(arr, n));
                    break;
                case 3:
                    System.out.println("The second largest element in the array : " + Arr.secondLargest(arr, n));
                    break;
                case 4:
                    System.out.println("Exiting program...");
                    break;
                default:
                    System.out.println("Invalid Choice...");
                    break;
            }
        } while (choice != 4);
        sc.close();
    }
}

File: Arrays/ArrOperations.cpp
#include <iostream>
using namespace std;

class ArrayOperations
{
public:
    // function to take input
    void read(int arr[], int n)
    {
        cout << "Enter the array elements : ";
        for (int i = 0; i < n; i++)
        {
            cin >> arr[i];
        }
    }

    // Left rotation of an array by one position
    void leftRotate(int arr[], int n)
    {
        int temp = arr[0];
        for (int j = 1; j < n; j++)
        {
            arr[j - 1] = arr[j];
        }
        arr[n - 1] = temp;

        // to print the array
        cout << "Array elements after rotation : ";
        for (int k = 0; k < n; k++)
        {
            cout << arr[k] << " ";
        }
        cout << endl;
    }
};

int main()
{
    ArrayOperations arr;
    int n;
    cout << "Enter the size of an array : ";
    cin >> n;

    int a[n];
    int choice;
    do
    {
        cout << "\nMENU\n";
        cout << "1. To read the array elements\n";
        cout << "2. Left rotation of an array by one bit\n";
        cout << "Enter your choice : ";
        cin >> choice;

        switch (choice)
        {
        case 1:
            arr.read(a, n);
            break;

        case 2:
            arr.leftRotate(a, n);
            break;

        default:
            cout << "Invalid choice\n";
            break;
        }
    } while (choice != 3); // Adjusted loop condition to match menu options

    return 0;
}
